libyul/optimiser/DataFlowAnalyzer.h

```diff
@@ -168,14 +154,24 @@ class DataFlowAnalyzer: public ASTModifier
 		Expression const& _expression
 	) const;
 
+	/// @returns true if the block is empty or the last statement has a
+	/// "flow out" control flow.
+	/// Note that this only considers the last statement. If there are
+	/// "leave", "break" or "continue" statements inside the block, this
+	/// has to be considered elsewhere.
+	bool hasFlowOutControlFlow(Block const& _block) const;
+
 	Dialect const& m_dialect;
 	/// Side-effects of user-defined functions. Worst-case side-effects are assumed
 	/// if this is not provided or the function is not found.
 	std::map<YulString, SideEffects> m_functionSideEffects;
+	std::map<YulString, ControlFlowSideEffects> m_controlFlowSideEffects;
 
 private:
 	struct State
 	{
+		/// Latest assignment counter per variable.
+		std::map<YulString, size_t> latestAssignment;
```

Comment:
I'd somehow feel better with a ``std::map<YulString, Statement const*> latestAssignment;`` or something like that - which would be enough, since we only need to check for equality and which wouldn't raise the question of whether ``<`` on the values has special significance or not...
But getting a nice suitable pointer in each case would be a hassle, so I guess a unique counter value is just fine...